libs/build_met.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash                                                                                                                              
+
+set -eux
+
+name="met"
+version=${1:-${STACK_met_version}}
+release_date=${2:-${STACK_met_release_date}}
+
+cd  ${HPC_STACK_ROOT}/${PKGDIR:-"pkg"}
+software=$name-$version.$release_date
+pkg_name=$name-$version
+url="https://github.com/dtcenter/MET/releases/download/v$version/$software.tar.gz"
+[[ -d $software ]] || ( $WGET $url; tar -xf $software.tar.gz )
+[[ -d $pkg_name ]] && cd $pkg_name || ( echo "$pkg_name does not exist, ABORT!"; exit 1 )
+
+
+# Hyphenated version used for install prefix
+compiler=$(echo $HPC_COMPILER | sed 's/\//-/g')
+mpi=$(echo $HPC_MPI | sed 's/\//-/g')
+
+export LMOD_EXACT_MATCH="no"
```

Comment:
This line does not belong in this script.  Please remove.

Reply:
@aerorahul, @kgerheiser told me to add this line.  Please see our correspondence via email below.

Julie:
> I was able to get down to the MET build by setting the build on NCO to "NO".  However, there seems to be a problem when loading the necessary modules for MET from build_met.sh.
> 
> From the build_stack.log file:
> ...
> + tar -xf met-9.1.1.20201118.tar.gz
> + [[ -d met-9.1.1 ]]
> + cd met-9.1.1
> ++ echo ips/18.0.1.163
> ++ sed 's/\//-/g'
> + compiler=ips-18.0.1.163
> + true
> + set +x
> Lmod has detected the following error: These module(s) exist but cannot be loaded as requested: "hdf5"
>    Try: "module spider hdf5" to see how to load the module(s).
> 
> I looked at the other build scripts and it looks like those modules are being loaded in the same way. 
> 
> For example, in build_netcdf.sh, I see:
> if $MODULES; then
>     set +x
>     source $MODULESHOME/init/bash
>     module load hpc-$HPC_COMPILER
>     [[ -z $mpi ]] || module load hpc-$HPC_MPI
>     module try-load szip
>     module load hdf5
> ...
> 
> From build_met.sh:
> if $MODULES; then
>    
>     set +x
>     source $MODULESHOME/init/bash
>     module load hpc-$HPC_COMPILER
>     module load python/3.6.3
>     module load hdf5
>  
> What am I doing wrong?  How can I get past this issue?
> 
> If you'd like to look at the entire log file, you can find it here on mars:
> /gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/hpc-stack/git/build_stack.attempt12.log
> 
> Thanks for any help you can give.

Kyle:
> You're on WCOSS, right? Put this in your environment: export LMOD_EXACT_MATCH="no"
> LMOD requires exact version on WCOSS and  the module is trying to load it without the version. 

Sure enough, adding that line fixed the problem.  If this needs to be taken out, I'll likely need some help from you all to get this working.